google-cloud-vision/test/google/cloud/vision/v1p1beta1/image_annotator_client_test.rb

```diff
@@ -0,0 +1,148 @@
+class MockGrpcClientStub_v1p1beta1
+
+  # @param expected_symbol [Symbol] the symbol of the grpc method to be mocked.
+  # @param mock_method [Proc] The method that is being mocked.
+  def initialize(expected_symbol, mock_method)
+    @expected_symbol = expected_symbol
+    @mock_method = mock_method
+  end
+
+  # This overrides the Object#method method to return the mocked method when the mocked method
+  # is being requested. For methods that aren't being tested, this method returns a proc that
+  # will raise an error when called. This is to assure that only the mocked grpc method is being
+  # called.
+  #
+  # @param symbol [Symbol] The symbol of the method being requested.
+  # @return [Proc] The proc of the requested method. If the requested method is not being mocked
+  #   the proc returned will raise when called.
+  def method(symbol)
+    return @mock_method if symbol == @expected_symbol
+
+    # The requested method is not being tested, raise if it called.
+    proc do
+      raise "The method #{symbol} was unexpectedly called during the " \
+        "test for #{@expected_symbol}."
+    end
+  end
+end
+
+class MockImageAnnotatorCredentials_v1p1beta1 < Google::Cloud::Vision::V1p1beta1::Credentials
+  def initialize(method_name)
+    @method_name = method_name
+  end
+
+  def updater_proc
+    proc do
+      raise "The method `#{@method_name}` was trying to make a grpc request. This should not " \
+          "happen since the grpc layer is being mocked."
+    end
+  end
+end
+
+describe Google::Cloud::Vision::V1p1beta1::ImageAnnotatorClient do
+
+  describe 'batch_annotate_images' do
+    custom_error = CustomTestError_v1p1beta1.new "Custom test error for Google::Cloud::Vision::V1p1beta1::ImageAnnotatorClient#batch_annotate_images."
+
+    it 'invokes batch_annotate_images without error' do
+      # Create request parameters
+      requests = []
+
+      # Create expected grpc response
+      expected_response = {}
+      expected_response = Google::Gax::to_proto(expected_response, Google::Cloud::Vision::V1p1beta1::BatchAnnotateImagesResponse)
+
+      # Mock Grpc layer
+      mock_method = proc do |request|
+        assert_instance_of(Google::Cloud::Vision::V1p1beta1::BatchAnnotateImagesRequest, request)
+        requests = requests.map do |req|
+          Google::Gax::to_proto(req, Google::Cloud::Vision::V1p1beta1::AnnotateImageRequest)
+        end
+        assert_equal(requests, request.requests)
+        OpenStruct.new(execute: expected_response)
+      end
+      mock_stub = MockGrpcClientStub_v1p1beta1.new(:batch_annotate_images, mock_method)
+
+      # Mock auth layer
+      mock_credentials = MockImageAnnotatorCredentials_v1p1beta1.new("batch_annotate_images")
+
+      Google::Cloud::Vision::V1p1beta1::ImageAnnotator::Stub.stub(:new, mock_stub) do
+        Google::Cloud::Vision::V1p1beta1::Credentials.stub(:default, mock_credentials) do
+          client = Google::Cloud::Vision.new(version: :v1p1beta1)
```

Comment:
This is going to fail. The constructor is now `Vision::ImageAnnotator.new`, not `Vision.new`.